src/app/public/src/modules/sidebar/sidebar-wrapper.component.ts

```diff
@@ -1,85 +1,72 @@
   styleUrls: ['./sidebar-wrapper.component.scss']
 })
-export class StacheSidebarWrapperComponent implements OnInit, OnDestroy, AfterViewInit {
+export class StacheSidebarWrapperComponent implements  OnDestroy, AfterViewInit {
   @Input()
   public sidebarRoutes: StacheNavLink[];
 
-  public sidebarOpen: boolean = false;
+  public isOpen: boolean = false;
 
   public sidebarLabel: string = 'Click to open sidebar';
 
-  private ngUnsubscribe: Subject<any> = new Subject();
+  public elementId = `stache-sidebar-content-panel-${(nextUniqueId++)}`;
+
+  private stacheContainers: HTMLElement[];
 
-  private stacheContainer: HTMLElement;
+  private mediaQuerySubscription: Subscription;
 
   constructor(
     private renderer: Renderer2,
-    private elementRef: ElementRef,
-    private omnibarService: StacheOmnibarAdapterService,
-    private windowRef: StacheWindowRef
+    private windowRef: StacheWindowRef,
+    private mediaQueryService: SkyMediaQueryService
   ) {
-    this.windowRef.onResize$
-      .takeUntil(this.ngUnsubscribe)
-      .subscribe(() => {
-        this.checkWindowWidth();
-        this.updateAriaLabel();
-      });
-  }
 
-  public ngOnInit(): void {
-    this.setTopAffix();
-    this.checkWindowWidth();
-    this.updateAriaLabel();
+    this.mediaQuerySubscription = this.mediaQueryService
+     .subscribe((args: SkyMediaBreakpoints) => {
+       this.isOpen = (args <= SkyMediaBreakpoints.sm);
+       this.toggleSidebar();
+     });
   }
 
   public ngAfterViewInit(): void {
-    this.stacheContainer = this.windowRef.nativeWindow.document.querySelector('.stache-container');
-    if (this.stacheContainer) {
-      this.renderer.addClass(this.stacheContainer, CONTAINER_SIDEBAR_CLASSNAME);
-    }
+    this.stacheContainers = this.windowRef.nativeWindow.document.querySelectorAll('.stache-container');
+    this.addClassToContainers();
   }
 
-  public setTopAffix(): void {
-    let omnibarHeight = this.omnibarService.getHeight();
-    let wrapperElement = this.elementRef.nativeElement.querySelector('.stache-sidebar-wrapper');
-    this.renderer.setStyle(wrapperElement, 'top', `${omnibarHeight}px`);
+  public toggleSidebar(): void {
+    this.isOpen = !this.isOpen;
   }
 
   public ngOnDestroy(): void {
-    if (this.stacheContainer) {
-      this.renderer.removeClass(this.stacheContainer, CONTAINER_SIDEBAR_CLASSNAME);
-    }
-    this.ngUnsubscribe.next();
-    this.ngUnsubscribe.complete();
+    this.removeClassFromContainers();
+    this.mediaQuerySubscription.unsubscribe();
   }
 
-  public toggleSidebar(): void {
-    this.sidebarOpen = !this.sidebarOpen;
-    this.updateAriaLabel();
-  }
-
-  public updateAriaLabel(): void {
-    this.sidebarLabel = this.sidebarOpen ? 'Click to close sidebar' : 'Click to open sidebar';
+  private addClassToContainers(): void {
+    if (this.stacheContainers && this.stacheContainers.length) {
+      Array.prototype.forEach.call(this.stacheContainers, (container: HTMLElement) => {
```

Comment:
Curious if there's a way to toggle the appropriate styles for the `.stache-container` in the style sheet, instead of complicating the component logic?

Reply:
I think the only way we could handle that is if we used the component to add a class to the body tag and then styled that way.